transport/http/inbound.go

```diff
@@ -51,6 +51,23 @@ func Mux(pattern string, mux *http.ServeMux) InboundOption {
 	}
 }
 
+// FallbackHandler specifies an http.Handler that will be executed when
+// the provided accepts function returns false. If accepts returns true,
+// then the request will be executed regularly.
+//
+// This allows existing HTTP services to add YARPC procedures to a shared
+// mux by testing if the request should be routed to YARPC or their existing
+// handler.
+//
+// For example, one might test for the presence of RPC-Encoding to determine if
+// the request should go to YARPC or their existing handler.
+func FallbackHandler(accepts func(req *http.Request) bool, handler http.Handler) InboundOption {
```

Comment:
Having two inputs here feels weird - for example, what if a user makes a mistake and one of them is nil? `accepts` is weird because when I see this, I feel like `accepts` should mean that `handler` is executed, until I read the docs. It also just feels like they could somehow be combined and would be safer.

Couldn't we just execute this if we can't find a procedure, ie build this into dispatching? Like why do we need to specify our own logic for accepts/why can't we just give a handler? It's not really a fallback handler in this sense, it's more like a "check if you wanna execute this thing first, if not, do the normal logic".

One other thing - adding this messes with eventual combining of http and grpc a little bit, but not completely.

Reply:
Some good points here.

* I renamed the `accepts` function to `checks` and inverted the bool check so that the `handler` is executed when `checks` returns `true`.

Reply:
Can you elaborate on how this messes with combining http and grpc?

Reply:
Just added validation so the option will err when taking nils - that should make it more straightforward to use.